src/v2/guide/typescript.md

```diff
@@ -0,0 +1,104 @@
+---
+title: TypeScript
+type: guide
+order: 25
+---
+
+If you develop a large application, using a language having static type checking would be helpful. If your code seems to cause unexpected behavior, it provides useful errors before you run it on browsers. You can easily use [TypeScript](https://www.typescriptlang.org/) (one of popular AltJS having static type checking) with Vue since Vue officially provides its declaration files.
```

Comment:
> If your code seems to cause unexpected behavior

This sentence does not make much sense to me.... Maybe `If you have unexpected typos in your code` is better?